rotkehlchen/tests/api/blockchain/test_evm.py

```diff
@@ -284,3 +289,41 @@ def test_add_multievm_accounts(rotkehlchen_api_server):
     assert result == [
         {'address': common_account, 'label': label, 'tags': ['metamask']},
     ]
+
+
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY]])
+def test_evm_account_deletion_does_not_wait_for_pending_txn_queries(rotkehlchen_api_server) -> None:  # noqa: E501
+    """
+    Test that if transactions for an address is being queried and account removal is issued,
+    the transactions querying greenlet is killed & the account is deleted subsequently.
+    """
+    # make an async query of all transactions.
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionsresource',
+        ), json={
+            'async_query': True,
+            'only_cache': False,
+            'limit': 1000,
+            'address': ADDY,
+            'evm_chain': 'ethereum',
+        },
+    )
+    assert_ok_async_response(response)
+
+    # now delete the address in sync -- and see that it does not take time.
+    seconds = 2
+    with gevent.Timeout(seconds):
```

Comment:
I would also somehow check whether the task was indeed killed.

Also I think will make sense to explicitly check if the locks were acquired and then released

Reply:
done.